### ISSUE TITLE
Lexer: Postfix conditionals in object literals

### ISSUE THREAD

**Issue by couchand** [jashkenas/coffeescript#4544]
*This is a repost of #3534, with additional detail since @GeoffreyBooth closed that issue as unclear. 
 Some valuable discussion, including potential workarounds, can be found on the original issue.*

There seems to be some inconsistency in the handling of postfix conditionals within nested object literals. For instance:

```coffeescript
# this compiles correctly
one =
  foo:
    baz: 1337
    bar: 42 if qux
```

The output, as one would expect, is:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var one;

  one = {
    foo: {
      baz: 1337,
      bar: qux ? 42 : void 0
    }
  };

}).call(this);
```

However, this (apparently equivalent code) fails to compile at all (`error: unexpected newline`):

```coffeescript
# this does not compile
two =
  foo:
    bar: 42 if qux
    baz: 1337
```

The expected behavior is that this would compile to the following:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var two;

  two = {
    foo: {
      bar: qux ? 42 : void 0,
      baz: 1337
    }
  };

}).call(this);
```

The general principle applied here is that swapping two lines of input at the same nesting level of an object should result in those lines simply swapped in the output.

When adding intervening function calls, the results are even more confusing.  For example:

```coffeescript
# this compiles correctly
three =
  foo: call
    baz: 1337
    bar: 42 if qux
```

This compiles to the expected:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var three;

  three = {
    foo: call({
      baz: 1337,
      bar: qux ? 42 : void 0
    })
  };

}).call(this);
```

However, the very similar

```coffeescript
# this compiles incorrectly
four =
  foo: call
    bar: 42 if qux
    baz: 1337
```

Compiles to the strange:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var four;

  four = {
    foo: call(qux ? {
      bar: 42
    } : void 0, {
      baz: 1337
    })
  };

}).call(this);
```

Whereas one would expect it to likewise result in a simple swap of the nested object lines:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var four;

  four = {
    foo: call({
      bar: qux ? 42 : void 0,
      baz: 1337
    })
  };

}).call(this);
```

In particular, the results as generated seem to break the expectation that keys at the same nesting level in the source will be put into a single object of the output.

Finally, this similar example, which seems to have pretty obviously wrong semantics:

```coffeescript
five =
  foo: call
    bar: 1337
  baz: 42 if qux
```

This compiles to:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var five;

  five = qux ? {
    foo: call({
      bar: 1337
    }),
    baz: 42
  } : void 0;

}).call(this);
```

But one would expect it to compile to:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var five;

  five = {
    foo: call({
      bar: 1337
    }),
    baz: qux ? 42  : void 0
  };

}).call(this);
```

One might try to get the expected semantics by reversing the order of the `foo` and `baz` keys, like so:

```coffeescript
six =
  baz: 42 if qux
  foo: call
    bar: 1337
```

But of course, as with example two above, this results in an `error: unexpected newline`.  In case it wasn't clear from the previous examples, this should compile to:

```js
// Generated by CoffeeScript 1.12.5
(function() {
  var five;

  five = {
    baz: qux ? 42  : void 0,
    foo: call({
      bar: 1337
    })
  };

}).call(this);
```

Hopefully this provides enough details to make clear what the issue is as well as the expected behavior, but if not, feel free to ask any clarifying questions!

**Comment by helixbass** [comment]
@couchand check out #4554 for an explanation of why it was behaving this way and a fix

**Comment by GeoffreyBooth** [comment]
Fixed via #4544.
